### PR TITLE
Humans_Engine: Avalue analysis improvements

### PR DESCRIPTION
**Pull request by rolyhudson**
### NOTE: Depends on 
https://github.com/BHoM/BHoM/pull/1231

   
### Issues addressed by this PR

Closes https://github.com/BHoM/BHoM_Engine/issues/2442
Review required for the [documentation of method](https://github.com/BHoM/documentation/wiki/BHoM-View-quality-conventions#measure-avalues) intended for non-technical audience.

### Test files

[Test file](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/BHoM_Engine/Humans_Engine/%232449-AValueAnalysisImprovements/A_valueexplorer.gh?csf=1&web=1&e=nNW1zS)
[Installer for this PR](https://burohappold.sharepoint.com/:u:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/BHoM_Engine/Humans_Engine/%232449-AValueAnalysisImprovements/Humans_Engine-%232442-AValueAnalysisImprovements.msi?csf=1&web=1&e=8uMGJy)

### Changelog

- Removed hard coding of view cones and spectator head outlines
- Removed use of the Theatron.ActivityArea
- Ensured the Avalue Settings terminology correspond with CvalueSettings
- Improved descriptions
- Added missing descriptions
- Removed legacy debug code

### CODE REVIEW

**Comment by rolyhudson** [comment]
@BHoMBot check installer

**Comment by rolyhudson** [comment]
@BHoMBot check installer

**Comment by rolyhudson** [comment]
@FraserGreenroyd I am hoping to take advantage of the awesome installer generator for user reviews on this one 🚀 

Following @IsakNaslundBh's  notes from a few days ago... 
I have the following folder structure set up:
[02_Current\12_Scripts\02_Pull Request\BHoM\BHoM_Engine\Humans_Engine\#2449-AValueAnalysisImprovements](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/02_Pull%20Request/BHoM/BHoM_Engine/Humans_Engine/%232449-AValueAnalysisImprovements?csf=1&web=1&e=uyCpgr)
where #2449 is this PR number and 
_AValueAnalysisImprovements_ is the last part of the branch description after the hyphen

I ran check installer once and but had the PR number wrong in the folder name 🙃 
Now I have updated that but...
I can't seem to trigger another installer check.

When you have a second (during the normal working hours) please could you have a quick look and let me know what I have botched with this? Cheers ⚡

**Comment by FraserGreenroyd** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@rolyhudson all sorted, I think BHoMBot was having a little snooze after working hard on my null checking all day! :smile:

**Comment by rolyhudson** [comment]
@FraserGreenroyd Love it! Installer artefact ready to go 🚀. This will be a huge help in getting more eyes onto these PRs.

**Comment by rolyhudson** [comment]
@BHoMBot check compliance

**Comment by rolyhudson** [comment]
@BHoMBot check versioning

**Comment by rolyhudson** [comment]
@BHoMBot check installer

**Comment by rolyhudson** [comment]
@BHoMBot check installer

**Comment by rolyhudson** [comment]
Noting here that following walk through sessions - change requested to default sightline to the spectators neutral sightline.
Retaining option for specifying focal point for custom use cases.

**Comment by rolyhudson** [comment]
@BHoMBot check installer

**Comment by rolyhudson** [comment]
@BHoMBot check installer

File: Humans_Engine/Query/CvalueAnalysis.cs [hunk 1]
```diff
@@ -82,8 +82,10 @@ private static List<Cvalue> EvaluateCvalue(Audience audience, CvalueSettings set
             if (audience.Spectators.Count == 0)
                 return results;
 
-            KDTree<Spectator> spectatorTree = SetKDTree(audience);
+            SetKDTree(audience);
```
Comment (michaelhoehn) [inline Humans_Engine/Query/CvalueAnalysis.cs:85]: I've not yet tested, but wanted to get a couple of comments noted. I'm really liking the use of KD Trees here and think there's a case for broader use throughout the BHoM, specifically space planning at a multitude of scales. I do think this warrants at least another issue raised to introduce this functionality a bit more formally.

**Review by jdarcyBHE — DISMISSED** [review]
Test file runs and updates have been implemented, documentation for A value reviewed no comments either.

File: Humans_Engine/Query/CvalueAnalysis.cs (hunk 1 above)
Comment (rolyhudson) [reply Humans_Engine/Query/CvalueAnalysis.cs:85]: @michaelhoehn noted and will raise an issue. Do you have a specific use case / project that would benefit from a more generic KD Tree. Also see the user review from Joe. This one is now ready for your scrutiny 😄

**Comment by rolyhudson** [comment]
@BHoMBot check required

**Comment by rolyhudson** [comment]
@BHoMBot fix project file ref. 2694570840

**Comment by rolyhudson** [comment]
@BHoMBot check required

**Comment by FraserGreenroyd** [comment]
@rolyhudson can you confirm the reasoning for requesting this dispensation on this occasion?

**Comment by rolyhudson** [comment]
@FraserGreenroyd as far as I can see all the null-handling errors are from outside the scope of this PR. If any relate directly to files changed in this PR I am happy to deal with them.
The null-handling report appears to terminate mid-line so if I should check else where for a full version please can you let me know.

File: Humans_Engine/Query/CvalueAnalysis.cs (hunk 1 above)
Comment (michaelhoehn) [reply Humans_Engine/Query/CvalueAnalysis.cs:85]: Hey @rolyhudson I'm thinking about partitioning floorplates based on predefined constraints. For example, developers will frequently have a requirement to fit _n_ 1BR at 300sf, _n_ 2BR at 800sf, and so on. I imagine KD trees can be used to quickly partition a given space. Pair this with some controlling parameters (a seed for an easy example) and you can quickly output some layouts.

**Review by michaelhoehn — COMMENTED** [review]
@rolyhudson I've reviewed the code and performed a functionality review and believe it's ready to go generally. I'd recommend addressing the very minor compliance notes prior to merge. I'll leave this as a comment for now to allow you to address the compliance notes. I'm happy to approve when you're happy.

**Comment by rolyhudson** [comment]
@BHoMBot check compliance

**Comment by rolyhudson** [comment]
@BHoMBot check required

**Comment by rolyhudson** [comment]
@BHoMBot check required

**Comment by rolyhudson** [comment]
@BHoMBot check serialisation

**Comment by rolyhudson** [comment]
@BHoMBot check compliance

**Comment by rolyhudson** [comment]
@BHoMBot check documentation-compliance

**Comment by rolyhudson** [comment]
@BHoMBot check compliance

**Comment by rolyhudson** [comment]
@BHoMBot check null-handling

**Comment by rolyhudson** [comment]
@BHoMBot check versioning

**Comment by rolyhudson** [comment]
@BHoMBot check serialisation

**Comment by rolyhudson** [comment]
@BHoMBot check core

**Comment by rolyhudson** [comment]
@BHoMBot check installer

**Review by michaelhoehn — APPROVED** [review]
Still LGTM!

**Comment by michaelhoehn** [comment]
@BHoMBot check ready-to-merge

**Comment by rolyhudson** [comment]
Closing this to re open PR on branch from updated master

**Pull request closed without merging**